dbcon/joblist/tuplehashjoin.cpp

```diff
@@ -1549,15 +1548,16 @@ void TupleHashJoinStep::joinRunnerFcn(uint32_t threadID)
             if (local_inputRG.getRowCount() == 0)
                 continue;
 
-            joinOneRG(threadID, &joinedRowData, local_inputRG, local_outputRG, largeRow,
-                      joinFERow, joinedRow, baseRow, joinMatches, smallRowTemplates);
+            joinOneRG(threadID, joinedRowData, local_inputRG, local_outputRG, largeRow,
```

Comment:
I strongly disagree that we need to future encourage puting one RGData at a time into upstream DL. It is very, very inefficient. We won't address this with this patch but should have this in mind burrowing this operation deeper down the call stack.

Reply:
In most cases I would agree with you. But in the case of Cartesian join, or near Cartesian join or even somewhere in between (join on column with lots of non-unique values), joinOneRG can produce any number of output rows. It could be in the hundreds of thousands or millions, as is the test case here. When attempting to not do one RG at a time, just keep this in mind.